backend/src/main/java/org/example/backend/controller/BackendController.java

```diff
@@ -0,0 +1,41 @@
+package org.example.backend.controller;
+
+import org.example.backend.model.assemblyai.FileUploadRequest;
+import org.example.backend.service.AssemblyAiService;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+import reactor.core.publisher.Mono;
+import java.io.IOException;
+
+@RestController
+@RequestMapping("/api")
+public class BackendController {
+
+    private final AssemblyAiService assemblyAiService;
+
+    public BackendController(AssemblyAiService assemblyAiService) {
+        this.assemblyAiService = assemblyAiService;
+    }
+
+    // File wird bei AssemblyAi hochgeladen
+    @PostMapping("/upload")
+    public Mono<ResponseEntity<String>> uploadFile(@RequestParam("file") MultipartFile file) throws IOException {
+        // MultipartFile in Record umwandeln
+            FileUploadRequest fileUploadRequest = new FileUploadRequest(
+                    file.getOriginalFilename(),
+                    file.getBytes()
+            );
+            return assemblyAiService.uploadFile(fileUploadRequest)
```

Comment:
I think the IOException is only handled in a Global Exception handler, so you might need to consider adding more error handling at some point.


***For example add error handling for cases:***

- The file might be too large.
- The file might be in an unsupported format.
- The AssemblyAI API could fail or be unreachable.
- There could be an exception in WebClient when sending the request.